src/app/games/rectanglz/RectanglzMoveGenerator.ts

```diff
@@ -0,0 +1,34 @@
+import { MoveGenerator } from 'src/app/jscaip/AI/AI';
+import { RectanglzMove } from './RectanglzMove';
+import { RectanglzConfig, RectanglzNode, RectanglzRules } from './RectanglzRules';
+import { RectanglzState } from './RectanglzState';
+import { Player } from 'src/app/jscaip/Player';
+import { MGPMap } from 'src/app/utils/MGPMap';
+import { Coord } from 'src/app/jscaip/Coord';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+
+export class RectanglzMoveGenerator extends MoveGenerator<RectanglzMove, RectanglzState, RectanglzConfig> {
+
+    public rules: RectanglzRules = RectanglzRules.get();
+
+    public override getListMoves(node: RectanglzNode, config: MGPOptional<RectanglzConfig>): RectanglzMove[] {
+        const player: Player = node.gameState.getCurrentPlayer();
+        const jumps: RectanglzMove[] = [];
+        const duplicationMap: MGPMap<Coord, RectanglzMove> = new MGPMap();
+        for (const coordAndContent of node.gameState.getCoordsAndContents()) {
+            if (coordAndContent.content.equals(player)) {
+                const coordMoves: RectanglzMove[] =
+                    this.rules.getPossiblesMoves(node.gameState, coordAndContent.coord, config);
+                const coordJumps: RectanglzMove[] = coordMoves.filter((m: RectanglzMove) => m.isJump());
```

Comment:
plutôt que traverser la liste deux fois (une fois pour jumps, une fois pour duplications), tu peux faire une seule traversée (soit un reduce, soit un for dans une fonction séparée pour splitter, soit tu fais un for ici et dedans un if pour gérer les deux cas)